### PR TITLE
WIP: feat: Dockerize test

### PR DESCRIPTION
**Pull request by rpappalax**
r? @pdehaan

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+version: 2
```
Comment (rpappalax) [inline .circleci/config.yml:1]: remove this after proof of concept.  we'll be building directly on Jenkins instead

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.DS_Store
+*.sw[op]
+*.pyc
+__pycache__
+.cache
+.tox
+*.json
+*.asc
+log.txt
+run.sh
```
Comment (pdehaan) [inline .gitignore:10]: We're going to move the build.sh and [uncommitted] run.sh files into the new and improved README.md file. 🎉

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
+FROM ubuntu:16.04 
+USER root
+
+WORKDIR /tests
+
+ARG STAGE_CONFIG="{'zut':'allures'}"
```
Comment (pdehaan) [inline Dockerfile]: Better default? empty object? poop emoji?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+ENV MOZ_HEADLESS="1"
+ENV RUST_BACKTRACE="full"
```
Comment (pdehaan) [inline Dockerfile]: Is this [or should this be] configurable? I assume this is the reason for the vvvery vvverbose output when I run against stage.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+RUN apt-get -y update && \
+    apt-get install -y software-properties-common &&\
+    apt-get install -y wget unzip && \
+    apt-get install -y libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev  && \
```
Comment (pdehaan) [inline Dockerfile]: Should we merge the 3 `apt-get install -y` commands (L12-14) into a single command? Not sure we'd see a perf difference either way, so maybe keeping separate is more preferable.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    cd /usr/src
+
+RUN apt-get install -y libgtk-3-0 libdbus-glib-1-2 libgtk-3-dev libglib2.0-dev
```
Comment (pdehaan) [inline Dockerfile]: Should this be moved up adjacent to L14, or is there significance to doing the previous `cd /usr/src` on L15?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+#TODO: do we need all of these or just one?
+RUN rm -rf /var/lib/apt/lists/* &&  \
+    apt-get -y --purge dist-upgrade && \
+    apt-get -y --purge autoremove -qq && \
+    apt-get clean -y
```
Comment (pdehaan) [inline Dockerfile]: Based on our 2 second searches, it sounds like lines 21+23 _may_ be unnecessary? Not sure if we want to save the installed dependency names into a `$buildDeps` into a variable so we can explicitly `apt-get purge -y --auto-remove $buildDeps -qq` (or whatever).

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+clear
+docker build -t firefoxtesteng/tps2 .
```
Comment (pdehaan) [inline build.sh:2]: Did we want to kill this file and move it into README.md?

File: Dockerfile (hunk 4 above)
Comment (rpappalax) [reply Dockerfile]: yea, let's drop this

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    rm master.zip
+
+RUN wget -O firefox-nightly.tar.bz2 'https://download.mozilla.org/?product=firefox-nightly-latest-ssl&os=linux64&lang=en-US' && \
+    tar xjf firefox-nightly.tar.bz2 && \
+    mv -i firefox firefox-nightly && \
+    rm firefox-nightly.tar.bz2
+
+#TODO: confirm, do we need virtualenv at all?
+RUN cd /mnt/extra/mozilla-central/testing/tps && \
+    ./create_venv.py /tests/tps-stage
```
Comment (rpappalax) [inline Dockerfile]: need to parameterize "tps-stage" to accomodate variable test ENV (stage or prod)

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    tar xjf firefox-nightly.tar.bz2 && \
+    mv -i firefox firefox-nightly && \
+    rm firefox-nightly.tar.bz2
+
+#TODO: confirm, do we need virtualenv at all?
+RUN cd /mnt/extra/mozilla-central/testing/tps && \
+    ./create_venv.py /tests/tps-stage 
+
+CMD . /tests/tps-stage/bin/activate && \
+    echo ${STAGE_CONFIG} > stage-config.json &&\
```
Comment (rpappalax) [inline Dockerfile:46]: need to parameterize "tps-stage" to accomodate variable test ENV (stage or prod)

File: Dockerfile [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    /tests/tps-stage/bin/runtps --debug --binary="/tests/firefox-nightly/firefox" --configfile=/tests/stage-config.json
```
Comment (rpappalax) [inline Dockerfile:47]: need to parameterize "tps-stage" to accomodate variable test ENV (stage or prod)

**Pull request closed without merging**